nerfstudio/viewer/server/viewer_param.py

```diff
@@ -1,27 +1,57 @@
 from __future__ import annotations
 
-from typing import Any, Dict, List, Optional, Tuple
+from abc import abstractmethod
+from typing import Any, Callable, Dict, List, Optional, Tuple
 
 from nerfstudio.viewer.viser import GuiHandle, ViserServer
 
 
-class ViewerParameter:
-    def __init__(self, name: str, default_value, min_value=None, max_value=None):
+class ViewerElement:
+    def __init__(self, name: str, disabled=False):
         self.name = name
-        self.param_type = type(default_value)
-        self.cur_value = default_value
         self.gui_handle = None
+        self.disabled = disabled
 
-    def create_gui_element(self, viser_server: ViserServer) -> GuiHandle:
+    @abstractmethod
+    def _create_gui_handle(self, viser_server: ViserServer):
+        """
+        Returns the GuiHandle object which actually controls the parameter in the gui
+        """
+        raise RuntimeError("Cannot instantiate ViewerElement directly")
+
+    @abstractmethod
+    def install(self, viser_server: ViserServer):
+        raise RuntimeError("Must implement install() for a ViewerElement")
+
+
+class ViewerButton(ViewerElement):
+    def __init__(self, name: str, callable: Callable, disabled=False):
+        super().__init__(name, disabled=disabled)
+        self.fn = callable
+
+    def _create_gui_handle(self, viser_server: ViserServer):
```

Comment:
We should probably add return annotations, eg `def _create_gui_handle(self, viser_server: ViserServer) -> None:`

Reply:
true